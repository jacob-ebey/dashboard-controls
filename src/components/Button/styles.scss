@import '../colors.scss';

$activeColor: $color-primaryExtraDark;

.primaryButton {
  display: inline-block;
  padding: 10px;
  color: white;
  background-color: $color-primary;
  border: none;

  &:hover {
    background-color: $color-primaryDark;
  }
  &:active {
    background-color: $activeColor;
  }

  &.active {
    background-color: $activeColor;
  }
}
