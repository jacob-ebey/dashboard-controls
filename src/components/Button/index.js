import React from 'react';

import styles from './styles.scss';

export default class Component extends React.Component {
  render() {
    const { className, children, ...rest } = this.props;
    return (
      <button
        className={`${styles.primaryButton} ${className || ''}`}
        {...rest}
      >
        {children}
      </button>
    );
  }
}
