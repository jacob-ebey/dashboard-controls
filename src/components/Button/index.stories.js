import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Button from './index';

storiesOf('Button', module)
  .add('default', () => (
    <Button onClick={action('Button: onClick')}>Hello, World</Button>
  ));
