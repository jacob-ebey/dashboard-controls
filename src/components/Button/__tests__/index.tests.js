import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Button from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('components/Button - index.js', () => {
  const createProps = () => ({
    children: 'Hello, World',
  });

  test('Renders without error', () => {
    const props = createProps();

    const wrapper = shallow(<Button {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});