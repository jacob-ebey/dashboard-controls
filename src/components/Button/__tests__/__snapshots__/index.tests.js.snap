// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button - index.js Renders without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component>
    Hello, World
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "Hello, World",
      "className": "primaryButton ",
    },
    "ref": null,
    "rendered": "Hello, World",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Hello, World",
        "className": "primaryButton ",
      },
      "ref": null,
      "rendered": "Hello, World",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
